healthcheck_manipulation.cpp

```diff
@@ -0,0 +1,82 @@
+#include "control_msgs/msg/dynamic_joint_state.hpp"
+#include "fstream"
+#include "nav_msgs/msg/odometry.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "vector"
+
+using namespace std::chrono_literals;
+
+#define LOOP_PERIOD 2s
+#define MSG_VALID_TIME 5s
+
+std::chrono::steady_clock::time_point last_odom_msg_time;
+std::chrono::steady_clock::time_point last_join_msg_time;
+
+void write_health_status(const std::string &status) {
+  std::ofstream healthFile("/health_status.txt");
+  healthFile << status;
+}
+
+void odom_callback(const nav_msgs::msg::Odometry::SharedPtr msg) {
+  last_odom_msg_time = std::chrono::steady_clock::now();
+}
+
+void joint_callback(const control_msgs::msg::DynamicJointState::SharedPtr msg) {
+
+  std::vector<std::string> expected_joint_names = {
+      "fl_wheel_joint",
+      "fr_wheel_joint",
+      "rl_wheel_joint",
+      "rr_wheel_joint",
+      "gripper_left_joint",
+      "joint1",
+      "joint2",
+      "joint3",
+      "joint4",
+      "imu",
+  };
+
+  for (const auto &joint_name : msg->joint_names) {
+    if (std::find(expected_joint_names.begin(), expected_joint_names.end(),
+                  joint_name) == expected_joint_names.end()) {
+      return;
```

Comment:
What does it happen then the function returns here?
The elapsed time is infinity?

Reply:
then value `last_odom_msg_time` will not be updated, so `healthy_check` detect that last msg is too old and return unhealthy